### PR TITLE
Simplify maybe_initialize_jax_distributed_system()

### PR DESCRIPTION
**Pull request by SurbhiJainUSC**
# Description

This PR refactors the JAX distributed initialization process by simplifying the `maybe_initialize_jax_distributed_system()` function.

- **Why is this change being made:** To improve code readability.
- **The problem being solved:** The previous implementation of the distributed setup had unnecessary complexity and boilerplate. 
- **Why this is a good solution:** It consolidates the logic for checking and initializing the JAX distributed system, making the initialization flow easier to maintain.

Fixes: b/422838626

# Tests

CI tests

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: tests/unit/max_utils_test.py [hunk 1]
```diff
@@ -344,5 +344,122 @@ def test_regular_shape_unpadded(self):
     self.assertEqual(padding_amount, target_padding_amount)
 
 
+class TestMaybeInitializeJaxDistributedSystem(unittest.TestCase):
```
Comment (gobbleturk) [inline tests/unit/max_utils_test.py:347]: there may be weird interactions with pytest, - it looks like this is heavily mocked, so jax.distributed.initialize is never really called? Could you include a comment if this is the case. If you don't mock I am not sure this will work

**Review by gobbleturk — APPROVED** [review]

File: tests/unit/max_utils_test.py (hunk 1 above)
Comment (SurbhiJainUSC) [reply tests/unit/max_utils_test.py:347]: Yes, `jax.distributed.initialize` is completely mocked out here. I have added a comment to the test class explicitly calling this out to avoid any future confusion.

**Review by bvandermoon — APPROVED** [review]
